porch/controllers/rootsyncsets/pkg/controllers/rootsyncset/controller.go

```diff
@@ -176,25 +403,46 @@ func (r *RootSyncSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.RootSyncSet{}).
+		Watches(
+			&source.Channel{Source: r.channel},
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				var rssName string
+				var rssNamespace string
+				if o.GetLabels() != nil {
+					rssName = o.GetLabels()[rootSyncSetNameLabel]
```

Comment:
Nit: you don't have to check if it's nil, you can read a nil map (you just can't write to it))